Minesweeper.py

```diff
@@ -178,91 +170,82 @@ def get_remaining_flags(self):
                             self.tiles.flag(mouse_pos)
-                # DHB: the main loop logic and the draw code should be separated, by moving all draw calls into self.draw()
+
                 self.tiles.draw()
                 self.draw()
 
-            if self.game_won and not self.game_won_message_printed:
+            if self.game_won:
                 self.game_won_screen.draw()
                 self.state = "game_won"
 
             if self.state == "game_over" or self.state == "game_won":
                 self.game_won_screen.handle_events(events)
 
-            # DHB: this is ugly in and of itself. Add comment or move to helper function
             if self.start_time != 0 and not self.game_over and not self.game_won:
                 self.time_elapsed = (pygame.time.get_ticks() - self.start_time) // 1000
 
             if self.state != "tutorial":
-            # Render and blit the clock text onto the screen
-                # DHB: the comment above was incorreclty indented
                 font = pygame.font.SysFont(None, 36)
                 clock_text = font.render(f"Time: {self.time_elapsed}", True, Color.RED)
-                win.blit(clock_text, (760, 885)) # Display the clock text at (10, 10) position on the screen
+                win.blit(clock_text, (760, 885))
 
-            # DHB; the end of the main loop should likely just call self.draw, which should do the background fill, call all the sub-draw-functions, and then flip
             pygame.display.flip()
 
-    def restart(self): # restarts the game
-        # DHB: the __init__ function should call the restart function, rather than duplicating the logic of setting up the state in that function and here
-        self.tiles = Tiles() # Reinitialize the Tiles object
+    def restart(self):
+        """restarts the game to the tutorial state when triggered
+        """
+        self.tiles = Tiles()
         self.game_over = False
         self.game_won = False
-        self.game_won_message_printed = False
-        self.state = "tutorial" # Reset the game state
+        self.state = "tutorial"
         self.start_time = 0
         self.time_elapsed = 0
-        # DHB: this adds self.tiles as a game object every time the game is restarted, which is unnecessary
-        self.add_game_object(self.tiles) # Add the Tiles object back to the game_objects list
-
-# Define the Field class to represent the play area
-class Field(object):
-    def draw(self):
-        # DHB: I'm pretty certain that this is dead code, i.e. it is never called by anything.
-        win.fill(Color.BACKGROUND)
 
-# Define the Tiles class to handle the game tiles
 class Tiles(object):
-    # DHB: constants should always be in upper case
+    """handles the game tiles
+    """
     num_rows = 16
     num_columns = 16
     TILE_SIZE = 52
     GAP = 3
     BUFFER = 5
 
     def __init__(self):
+        """ sets variables for use later"""
```

Comment:
too general comment, and the function initialises the instance of Tiles, not just "sets variables for use later".